### PR TITLE
feat(18772): use new /axis endpoint for MCDA

### PR DESCRIPTION
**Pull request by albaranau**
https://kontur.fibery.io/Tasks/Task/FE-Replace-api-bivariate_matrix-endpoint-with-new-one-for-MCDA-18772

## Summary by CodeRabbit

- **New Features**
  - Introduced functions to fetch multiple axes data and bivariate axes.
  - Updated `MCDAForm` to use the new `availableBivariateAxesAtom`.

- **Improvements**
  - Enhanced the filtering and mapping of axes data based on quality.
  - Improved error handling and data refetching capabilities.
  - Enhanced label formatting capabilities for bivariate axes with custom labels.

- **Dependencies**
  - Added new dependencies for `@reatom/async` and `@reatom/hooks` to support enhanced asynchronous operations and state management.

### CODE REVIEW

File: src/core/api/mcda.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { apiClient } from '~core/apiClientInstance';
+import type { AxisDTO } from '~features/bivariate_manager/types';
+
+export function getBivariateAxes(abortController?: AbortController) {
+  return apiClient.get<AxisDTO[]>('/axis', undefined, true, {
```
Comment (albaranau) [inline src/core/api/mcda.ts]: Right now `/axis` endpoint returns 204 No Body on production, so this PR shouldn't be merged until that's resolved

File: src/core/api/mcda.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { apiClient } from '~core/apiClientInstance';
+import type { AxisDTO } from '~features/bivariate_manager/types';
```
Comment (vkozio) [inline src/core/api/mcda.ts]: backend-related types (DTO's) for this endpoint should be in this file

**Review by vkozio — DISMISSED** [review]

File: src/core/api/mcda.ts (hunk 2 above)
Comment (albaranau) [reply src/core/api/mcda.ts]: It's not the only endpoint this type is currently used in (`/bivariate_matrix` also uses it still), so not sure I should move it out there yet.

Comment (vkozio) [reply src/core/api/mcda.ts]: we do not import from features/ in core/, also we have core/bivariate/ so maybe it's a good place to collect bivariate types used across app

Comment (albaranau) [reply src/core/api/mcda.ts]: Fixed

File: src/features/mcda/components/MCDAForm/index.tsx [hunk 3]
```diff
@@ -55,10 +56,11 @@ export function MCDAForm({
     },
     [],
   );
-  const [axisesResource] = useAtom(availableBivariateAxisesAtom);
+  const [axisesResource] = useAtom(availableBivariateAxesAtom);
+  // console.log({axisesResource});
```
Comment (Natallia-Harshunova) [inline src/features/mcda/components/MCDAForm/index.tsx]: commented console logs

File: src/features/mcda/components/MCDAForm/index.tsx [hunk 4]
```diff
@@ -55,10 +56,11 @@ export function MCDAForm({
     },
     [],
   );
-  const [axisesResource] = useAtom(availableBivariateAxisesAtom);
+  const [axisesResource] = useAtom(availableBivariateAxesAtom);
```
Comment (Natallia-Harshunova) [inline src/features/mcda/components/MCDAForm/index.tsx]: maybe rename it with axesResource

**Review by Natallia-Harshunova — APPROVED** [review]

File: src/features/mcda/components/MCDAForm/index.tsx (hunk 3 above)
Comment (albaranau) [reply src/features/mcda/components/MCDAForm/index.tsx]: Fixed

File: src/features/mcda/components/MCDAForm/index.tsx (hunk 4 above)
Comment (albaranau) [reply src/features/mcda/components/MCDAForm/index.tsx]: fixed

**Pull request merged**